tests/pyfunc/test_spark.py

```diff
@@ -120,6 +121,37 @@ def test_spark_udf(spark, model_path):
                 actual = list(new_df.select("prediction").toPandas()["prediction"])
                 assert expected == actual
 
+def test_spark_udf_autofills_no_arguments(spark, model_path):
```

Comment:
We may want to merge this test with `test_spark_udf_autofills_column_names_with_schema`